shimmingtoolbox/optimizer/lsq_optimizer.py

```diff
@@ -86,6 +93,31 @@ def _residuals_mae(self, coef, unshimmed_vec, coil_mat, factor):
         # MAE regularized to minimize currents
         return np.mean(np.abs(unshimmed_vec + np.sum(coil_mat * coef, axis=1, keepdims=False))) / factor + \
             (self.reg_factor * np.mean(np.abs(coef) / self.reg_factor_channel))
+            
+    def _residuals_grad(self, coef, unshimmed_vec, coil_mat, factor):
+        """ Objective function to minimize the mean squared error (MSE) and the signal loss function (gradient in z direction)
+
+        Args:
+            coef (numpy.ndarray): 1D array of channel coefficients
+            factor (float): Devise the result by 'factor'. This allows to scale the output for the minimize function to
+                            avoid positive directional linesearch
+        Returns:
+            numpy.ndarray: Residuals for least squares optimization -- equivalent to flattened shimmed vector
+        """
+        #print("w_signal_loss is: " + str(self.w_signal_loss) + "," + " epi_te is: " + str(self.epi_te) + " factor is" + str(factor))
+        nx,ny,nz,nc = np.shape(self.merged_coils)
+        shimmed = self.unshimmed + np.sum(self.merged_coils * np.tile(coef,(nx,ny,nz,1)),axis= 3) # need test
+        signal = 1
+        # if consider signal loss from x, y, and z
+        for i in range(0,3):
+            G = np.gradient(shimmed, axis = i)
```

Comment:
```suggestion
            G = np.gradient(shimmed, axis=i)
```

Reply:
Thanks! addresed this in https://github.com/shimming-toolbox/shimming-toolbox/commit/9c2ba526bef3781adfbd5b34d95d37b3e7f5c7da